src/main/java/com/eternalcode/discordapp/review/GitHubReviewUtil.java

```diff
@@ -0,0 +1,101 @@
+package com.eternalcode.discordapp.review;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public final class GitHubReviewUtil {
+
+    private static final String GITHUB_PULL_REQUEST_REGEX = "^https://github\\.com/([a-zA-Z0-9-_]+/[a-zA-Z0-9-_]+)/pull/([0-9]+)$";
+    private static final String GITHUB_PULL_REQUEST_TITLE_CONVENTION = "^(GH)-\\d+ .+$";
+
+    private static final OkHttpClient HTTP_CLIENT = new OkHttpClient();
+    private static final Gson GSON = new Gson();
+
+    private GitHubReviewUtil() {}
+
+    public static boolean isPullRequestUrl(String url) {
+        return url.matches(GITHUB_PULL_REQUEST_REGEX);
+    }
+
+    public static boolean isPullRequestTitleValid(String title) {
+        return title.matches(GITHUB_PULL_REQUEST_TITLE_CONVENTION);
+    }
+
+    public static List<String> getReviewers(String url, String githubToken) {
+        Request request = new Request.Builder()
+                .url(url)
+                .header("Authorization", "token" + githubToken)
+                .build();
+
+        try {
+            Response response = HTTP_CLIENT.newCall(request).execute();
+            String responseBody = response.body().string();
+
+            JsonObject json = GSON.fromJson(responseBody, JsonObject.class);
+            JsonArray requestedReviewers = json.getAsJsonArray("requested_reviewers");
+
+            List<String> reviewers = new ArrayList<>();
+            for (int i = 0; i < requestedReviewers.size(); i++) {
+                JsonObject reviewer = requestedReviewers.get(i).getAsJsonObject();
+                String reviewerLogin = reviewer.get("login").getAsString();
+                reviewers.add(reviewerLogin);
+            }
+
+            return reviewers;
+        }
+        catch (IOException exception) {
+            exception.printStackTrace();
+            return Collections.emptyList();
+        }
+    }
+
+    public static String getGitHubPullRequestApiUrl(String url) {
+        GitHubPullRequestInfo gitHubPullRequestInfo = new GitHubPullRequestInfo(url);
```

Comment:
no teraz to w sumie ta instancja useless więc można zrobić z tego utila :trollface: